File: src/day_1.py
import sys

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Not enough arguments")
        exit()

    pos = 50
    zero_count = 0

    with open(sys.argv[1]) as f:
        for line in f:
            clicks = int(line[1:])
            delta = -clicks if line[0] == "L" else clicks

            # Part 1
            # pos = (pos + delta) % 100
            # if pos == 0:
            #     zero_count += 1

            # Part 2
            start = pos
            pos = (pos + delta) % 100

            if pos == 0:
                zero_count += 1
            elif delta > 0 and pos < start:
                zero_count += 1
            elif start != 0 and delta < 0 and pos > start:
                zero_count += 1

            zero_count += clicks // 100

    print(zero_count)

File: src/Day1.java
void main(String[] args) throws Exception {
    if (args.length < 1) {
        IO.println("Not enough arguments");
        return;
    }
    String input = Files.readString(Paths.get(args[0]));

    int pos = 50;
    int zeroCount = 0;

    for (String line : input.split("\n")) {
        int clicks = Integer.parseInt(line.substring(1));
        int delta = (line.charAt(0) == 'L') ? -clicks : clicks;

        // Part 1
        // pos = Math.floorMod(pos + delta, 100);
        // if (pos == 0) zeroCount++;

        // Part 2
        int start = pos;
        pos = Math.floorMod(pos + delta, 100);

        if (pos == 0) {
            zeroCount++;
        } else if (delta > 0 && pos < start) {
            zeroCount++;
        } else if (start != 0 && delta < 0 && pos > start) {
            zeroCount++;
        }
        zeroCount += Math.floorDiv(clicks, 100);
    }

    IO.println(zeroCount);
}
